### PR TITLE
Mick 1.2.x

### PR DESCRIPTION
**Pull request by mick-stewart73**

### CODE REVIEW

File: ikasaneip/dashboard/war/src/main/java/org/ikasan/dashboard/notification/email/EmailNotifier.java [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+import org.ikasan.dashboard.notification.NotificationContentProducer;
+import org.ikasan.dashboard.notification.Notifier;
+import org.ikasan.monitor.notifier.EmailNotifierConfiguration;
+
+/**
+ * 
+ * @author Ikasan Development Team
+ *
+ */
+public class EmailNotifier implements Notifier
```
Comment (andrzej-majewski) [inline ikasaneip/dashboard/war/src/main/java/org/ikasan/dashboard/notification/email/EmailNotifier.java]: I know this is new Implementation of notifier. But we have implementation of Email producer cant we reuse it rather then create duplication?

File: ikasaneip/dashboard/war/src/main/java/org/ikasan/dashboard/ui/topology/component/BusinessStreamTab.java [hunk 2]
```diff
@@ -427,10 +441,199 @@ public void buttonClick(ClickEvent event)
...
+					
+					networkDiagram.addNode(node);
+					networkDiagram.addEdge(edge);					
+	            }
+            }
+            
+        });
+        
+
+        ScheduledExecutorService executor = Executors.newScheduledThreadPool(1);
```
Comment (drawson) [inline ikasaneip/dashboard/war/src/main/java/org/ikasan/dashboard/ui/topology/component/BusinessStreamTab.java]: This executor is never shut down. Thread leak, will stop classes unloading.

File: ikasaneip/dashboard/war/src/main/java/org/ikasan/dashboard/ui/topology/component/BusinessStreamTab.java [hunk 3]
```diff
@@ -427,10 +441,199 @@ public void buttonClick(ClickEvent event)
...
-		layout.addComponent(this.businessStreamTable);
+//    	layout.addComponent(controlsLayout);
+//		layout.addComponent(this.businessStreamTable);
+		layout.addComponent(getNetwrokDiagram());
+		
+		this.addComponent(layout);
+		this.setSizeFull();
+	}
+	
+	public DragAndDropWrapper getNetwrokDiagram()
```
Comment (drawson) [inline ikasaneip/dashboard/war/src/main/java/org/ikasan/dashboard/ui/topology/component/BusinessStreamTab.java]: typo in method name.... getNetwrokDiagram()

**Comment by drawson** [comment]
org.ikasan.dashboard.ui.Broadcast also has a static ExecutorService thats never shutdown. You probably want a hook in similar to TopologyStateCache.shutdown() and call it from WebAppStartStopListener

File: ikasaneip/dashboard/war/src/main/java/org/ikasan/dashboard/notification/email/EmailNotifier.java (hunk 1 above)
Comment (mick-stewart73) [inline ikasaneip/dashboard/war/src/main/java/org/ikasan/dashboard/notification/email/EmailNotifier.java]: I am going to leave it the way it is now, as this has been tested and signed off. Will look at the email producer at some time. I remember looking at it but decided to go this way but can't remember why.

File: ikasaneip/dashboard/war/src/main/java/org/ikasan/dashboard/ui/topology/component/BusinessStreamTab.java (hunk 2 above)
Comment (mick-stewart73) [inline ikasaneip/dashboard/war/src/main/java/org/ikasan/dashboard/ui/topology/component/BusinessStreamTab.java]: Thanks for reminding me of this. This tab is a work in progress which will not be displayed in production. I have removed the executor for now.

**Comment by mick-stewart73** [comment]
Thanks for pointing this out Darren. I have sorted out the Broadcaster.

**Comment by drawson** [comment]
Merged, lets check the 1.2.x branch to ensure it still build correctly - travis still reports failed build

**Pull request merged**